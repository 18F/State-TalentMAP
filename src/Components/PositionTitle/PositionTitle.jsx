import React, { Component } from 'react';
import FontAwesome from 'react-fontawesome';
import PropTypes from 'prop-types';
import { POSITION_DETAILS } from '../../Constants/PropTypes';

class PositionTitle extends Component { // eslint-disable-line
  render() {
    const { details, goBackLink } = this.props;
    return (
      <div className="position-details-header" style={{ overflow: 'hidden', backgroundColor: '#F2F2F2' }}>
        <div className="usa-grid" style={{ overflow: 'hidden' }}>
          <div className="usa-width-one-half">
            <div className="position-details-header-back">
              { goBackLink && // if goBackLink is defined, render...
              <div>
                <FontAwesome name="arrow-left" />
                &nbsp;
                {goBackLink}
              </div>
              }
            </div>
            <div className="position-details-header-title">
              <strong>Position Number: {details.position_number}</strong>
            </div>
            <p className="position-details-header-body">
              <strong>Description: </strong>
              Lorem Ipsum is simply dummy text of the printing and typesetting industry.
              Lorem Ipsum has been the industry standard dummy text
              ever since the 1500s, when an unknown printer
              took a galley of type and scrambled it to make...
          </p>
            <div className="usa-width-one-half position-details-header-body">
              <strong>Post website:</strong> <a href="https://www.state.gov">www.state.gov</a>
            </div>
            <div className="usa-width-one-half position-details-header-body">
              <strong>Point of Contact:</strong> <a href="tel:222-222-2222">222-222-2222</a>
            </div>
          </div>
        </div>
        <img
          className="position-details-header-image"
          alt="department of state seal"
          src="/assets/img/dos-seal-bw.png"
        />
      </div>
    );
  }
}

PositionTitle.propTypes = {
  details: POSITION_DETAILS,
  goBackLink: PropTypes.node.isRequired,
};

PositionTitle.defaultProps = {
  details: null,
};

export default PositionTitle;
