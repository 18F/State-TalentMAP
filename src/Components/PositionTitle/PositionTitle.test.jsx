import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import PositionTitle from './PositionTitle';

describe('PositionTitleComponent', () => {
  let wrapper = null;

  const detailsObject = {
    id: 6,
    grade: '05',
    skill: 'OFFICE MANAGEMENT (9017)',
    bureau: '150000',
    organization: 'FREETOWN SIERRA LEONE (FREETOWN)',
    position_number: '00003027',
    is_overseas: true,
    create_date: '2006-09-20',
    update_date: '2017-06-08',
    post: { id: 162, tour_of_duty: '2YRR', code: 'LT6000000', description: 'MASERU, LESOTHO', cost_of_living_adjustment: 0, differential_rate: 15, danger_pay: 0, rest_relaxation_point: 'London', has_consumable_allowance: false, has_service_needs_differential: false },
    languages: [{ id: 1, language: 'French (FR)', written_proficiency: '2', spoken_proficiency: '2', representation: 'French (FR) 2/2' }],
  };

  it('can receive props', () => {
    wrapper = shallow(
      <PositionTitle details={detailsObject} isLoading={false} hasErrored={false} />,
    );
    expect(wrapper.instance().props.details.id).toBe(6);
  });

  it('matches snapshot', () => {
    wrapper = shallow(
      <PositionTitle details={detailsObject} isLoading={false} hasErrored={false} />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('handles different props and different position objects', () => {
    Object.assign(detailsObject, { languages: [], post: null, is_overseas: false });

    wrapper = shallow(
      <PositionTitle details={detailsObject} isLoading={false} hasErrored={false} />,
    );
    expect(wrapper.instance().props.details.languages.length).toBe(0);
  });

  it('handles different types of position objects', () => {
    Object.assign(detailsObject, { languages: [], is_overseas: true });

    wrapper = shallow(
      <PositionTitle details={detailsObject} isLoading hasErrored={false} />,
    );
    expect(wrapper.instance().props.details.languages.length).toBe(0);
  });
});
