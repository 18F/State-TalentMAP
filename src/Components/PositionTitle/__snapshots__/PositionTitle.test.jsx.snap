// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PositionTitleComponent matches snapshot 1`] = `
<div
  className="position-details-header"
  style={
    Object {
      "backgroundColor": "#F2F2F2",
      "overflow": "hidden",
    }
  }
>
  <div
    className="usa-grid"
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <div
      className="usa-width-one-half"
    >
      <div
        className="position-details-header-back"
      >
        Back to Search Results
      </div>
      <div
        className="position-details-header-title"
      >
        <strong>
          Position Number: 
          00003027
        </strong>
      </div>
      <p
        className="position-details-header-body"
      >
        <strong>
          Description: 
        </strong>
        Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make...
      </p>
      <div
        className="usa-width-one-half position-details-header-body"
      >
        <strong>
          Post website:
        </strong>
         www.post.gov
      </div>
      <div
        className="usa-width-one-half position-details-header-body"
      >
        <strong>
          Point of Contact:
        </strong>
         222-222-2222
      </div>
    </div>
  </div>
  <img
    alt="department of state seal"
    className="position-details-header-image"
    src="/assets/img/dos-seal-bw.png"
  />
</div>
`;
